src/klein/test/util.py

```diff
@@ -32,14 +35,14 @@ def test_identicalEq(self):
         An object compares equal to itself using the C{==} operator.
         """
         o = self.anInstance()
-        self.assertTrue(o == o)
+        cast(SynchronousTestCase, self).assertTrue(o == o)
```

Comment:
it might be nicer to use a self type here:


```
def test_identicalEq(self: unittest.TestCase):
```

Reply:
https://mypy.readthedocs.io/en/latest/more_types.html#mixin-classes

Reply:
maybe something like this can work?

```python
class EqualityTestProtocol(Protocol, Generic[T, U], unittest.TestCase):
    def anInstance(self) -> T
    def anotherInstance(self) -> U: ...

class EqualityTestsMixin(Generic[T, U]):
    def test_identicalEq(self: EqualityTestProtocol[T, U]):
        o = self.anInstance()
        self.assertTrue(o == o)
```

Reply:
OK, check out 6d04119

Reply:
Well… this worked nicely for `mypy` but fails unit tests; looks like you can't inherit from both `Protocol` and `TestCase`.

Reply:
```
class EqualityTestProtocol(Protocol, Generic[T, U], unittest.TestCase if TYPE_CHECKING else object):
    def anInstance(self) -> T
    def anotherInstance(self) -> U: ...

class EqualityTestsMixin(Generic[T, U]):
    def test_identicalEq(self: EqualityTestProtocol[T, U]):
        o = self.anInstance()
        self.assertTrue(o == o)
```

Reply:
oh right looking at the error you don't have Protocol - you need it from typing_extensions

Reply:
```
from typing_extensions import Protocol

class TestCaseProtocol(Protocol):
    def asssertTrue(v: bool) -> None: ...
    def assertFalse(v: bool) -> None: ...

class EqualityTestProtocol(Protocol, Generic[T], TestCaseProtocol):
    def anInstance(self) -> T
    def anotherInstance(self) -> T: ...
```